addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxWsClient.java

```diff
@@ -258,16 +262,33 @@ boolean connect() {
             socket = new LxWebSocket();
             wsClient = new WebSocketClient();
 
+            String message = socket.httpGet(CMD_CFG_API);
```

Comment:
All the syncronized(state) statements are incorrect here and have threading issues in general.  You can enter any on of these synchronized statements holding a lock on a specific instance - but then another thread can set a new state (via setClientState()) and your lock does nothing at that point (because the method could be reentered and a lock will be checked/held on the new reference).  In general, you should be synchronizing on some 'final' variable that doesn't change
  
Same applies to the unsynchronized usage in setClientState (that should be protected as well)

Reply:
Oh, yes, it's a shame. Fixing...